test/binder/binder_test.cpp

```diff
@@ -152,7 +161,10 @@ TEST_F(BinderCorrectnessTest, SelectStatementTest) {
   }
 
   // Test select from different table instances from the same physical schema
-  binder.reset(new binder::BindNodeVisitor());
+  txn_manager.CommitTransaction(txn);
+
+  txn = txn_manager.BeginTransaction();
+  binder->SetTxn(txn);
```

Comment:
Can the binder ever run without a txn? If it always needs it, then we should make it a required argument to its constructor.